clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -456,6 +457,12 @@ protected void onJoinComplete(int generation,
 
         subscriptions.assignFromSubscribed(assignedPartitions);
 
+        if (pausedPartitions != null && !pausedPartitions.isEmpty())
```

Comment:
Why not always try to get from `subscriptions.pausedPartitions()`, than maintaining a local copy? The paused partitions could change at any time in between as the onJoinXXX may be triggered by the hb thread right?

Reply:
If we use `subscriptions.pausedPartitions()` in onJoinComplete to get the `pausedPartitions` after the last rebalance is completed, for the Eager protocol, the `pausedPartitions` have been cleaned up in onJoinPrepare.